build-logic/jvm/src/main/kotlin/gradlebuild.unittest-and-compile.gradle.kts

```diff
@@ -384,7 +384,9 @@ fun Test.configureJvmForTest() {
 fun Test.addOsAsInputs() {
     // Add OS as inputs since tests on different OS may behave differently https://github.com/gradle/gradle-private/issues/2831
     // the version currently differs between our dev infrastructure, so we only track the name and the architecture
-    inputs.property("operatingSystem", "${OperatingSystem.current().name} ${System.getProperty("os.arch")}")
+    if (name != "archTest") {
```

Comment:
❌ Let's not have the if here but from the caller. Whatever adds the arch test task should not call this method. This method should not know about architecture tests.

Reply:
This is done here:

https://github.com/gradle/gradle/blob/5c4bbe21616c12ccbdc79c4eb446e4e022f7d17d/build-logic/jvm/src/main/kotlin/gradlebuild.unittest-and-compile.gradle.kts#L426

Why do you think it is better to have the filter there?

Reply:
I would prefer to have the call to this method where we add the other test tasks. And then the code adding the architecture test would not have to do this. In other works, I think `addOsAsInputs()` does not belong to the generic test configuration.

If we can't do that now, moving it up to the caller at least makes it more visible that we are doing the wrong thing here. This general script should not know about a possible test called `archTest` that is added later by something else. We need to add a TODO for sure.